### PR TITLE
Use a single entry-point for mounting

### PR DESCRIPTION
**Pull request by tobsan**
See the commit about this. Also a commit removing some unused variables.

### CODE REVIEW

**Comment by tobsan** [comment]
Just realized I didn't update docs, hang on with merging.

File: libsoftwarecontainer/include/container.h [hunk 1]
```diff
@@ -219,6 +208,34 @@ class Container :
...
+     *
+     * Will create missing parent directories.
+     *
+     * @param pathOnHost The path to the directory that shall be bind mounted on the host system.
+     * @param pathInContainer Where to mount the file in the container.
+     * @param readonly Sets if the mount should be read only or read write
+     *
+     * @return SUCCESS if everything worked as expected, FAILURE otherwise.
+     */
+    ReturnCode bindMountFolderInContainer(const std::string &pathOnHost, const std::string &pathInContainer, bool readonly = true);
```
Comment (breakreturn) [inline libsoftwarecontainer/include/container.h]: Could we rename this to not use "folder"?

File: libsoftwarecontainer/src/container.cpp [hunk 2]
```diff
@@ -516,6 +516,24 @@ ReturnCode Container::destroy(unsigned int timeout)
     return ReturnCode::SUCCESS;
 }
 
+ReturnCode Container::bindMountInContainer(const std::string &pathOnHost,
```
Comment (breakreturn) [inline libsoftwarecontainer/src/container.cpp]: Throughout it's path-ON-host and path-IN-container. Any reason for them to be different w.r.t. "in" and "on"?

File: libsoftwarecontainer/include/container.h (hunk 1 above)
Comment (tobsan) [reply libsoftwarecontainer/include/container.h]: Gotcha, will change to directory.

File: libsoftwarecontainer/src/container.cpp (hunk 2 above)
Comment (tobsan) [reply libsoftwarecontainer/src/container.cpp]: no, I will streamline.

File: libsoftwarecontainer/component-test/softwarecontainerlib_unittest.cpp [hunk 3]
```diff
@@ -365,7 +365,7 @@ TEST_F(SoftwareContainerApp, TestFileMounting) {
     job1.start();
     ASSERT_EQ(job1.wait(), NON_EXISTENT);
 
-    ReturnCode result = getSc().getContainer()->bindMountFileInContainer(tempFilename, tempFilename, true);
+    ReturnCode result = getSc().getContainer()->bindMountInContainer(tempFilename, tempFilename, true);
```
Comment (thenor) [inline libsoftwarecontainer/component-test/softwarecontainerlib_unittest.cpp]: Is it not possible to write the result and assert as a one-liner?

Comment (thenor) [reply libsoftwarecontainer/component-test/softwarecontainerlib_unittest.cpp]: Comment is for all the below cases which are written in the same way.

**Review by DunderRoffe — APPROVED** [review]
Looks nice. Although I like "bindMount" better than "bindMountInContainer".

File: libsoftwarecontainer/component-test/softwarecontainerlib_unittest.cpp (hunk 3 above)
Comment (tobsan) [reply libsoftwarecontainer/component-test/softwarecontainerlib_unittest.cpp]: Hmmm yes, I'll see what I can do.

**Review by thenor — APPROVED** [review]

**Pull request merged**